catanatron_rust/src/map_instance.rs

```diff
@@ -0,0 +1,476 @@
+                );
+            } else if neighbor_direction == Direction::SouthEast {
+                nodes.insert(
+                    NodeRef::South,
+                    *neighbor_hexagon.nodes.get(&NodeRef::NorthWest).unwrap(),
+                );
+                nodes.insert(
+                    NodeRef::SouthEast,
+                    *neighbor_hexagon.nodes.get(&NodeRef::North).unwrap(),
+                );
+                edges.insert(
+                    EdgeRef::SouthEast,
+                    *neighbor_hexagon.edges.get(&EdgeRef::NorthWest).unwrap(),
+                );
+            } else if neighbor_direction == Direction::SouthWest {
+                nodes.insert(
+                    NodeRef::South,
+                    *neighbor_hexagon.nodes.get(&NodeRef::NorthEast).unwrap(),
+                );
+                nodes.insert(
+                    NodeRef::SouthWest,
+                    *neighbor_hexagon.nodes.get(&NodeRef::North).unwrap(),
+                );
+                edges.insert(
+                    EdgeRef::SouthWest,
+                    *neighbor_hexagon.edges.get(&EdgeRef::NorthEast).unwrap(),
+                );
+            } else if neighbor_direction == Direction::West {
+                nodes.insert(
+                    NodeRef::NorthWest,
+                    *neighbor_hexagon.nodes.get(&NodeRef::NorthEast).unwrap(),
+                );
+                nodes.insert(
+                    NodeRef::SouthWest,
+                    *neighbor_hexagon.nodes.get(&NodeRef::SouthEast).unwrap(),
+                );
+                edges.insert(
+                    EdgeRef::West,
+                    *neighbor_hexagon.edges.get(&EdgeRef::East).unwrap(),
+                );
+            } else if neighbor_direction == Direction::NorthWest {
+                nodes.insert(
+                    NodeRef::North,
+                    *neighbor_hexagon.nodes.get(&NodeRef::SouthEast).unwrap(),
+                );
+                nodes.insert(
+                    NodeRef::NorthWest,
+                    *neighbor_hexagon.nodes.get(&NodeRef::South).unwrap(),
+                );
+                edges.insert(
+                    EdgeRef::NorthWest,
+                    *neighbor_hexagon.edges.get(&EdgeRef::SouthEast).unwrap(),
+                );
+            } else if neighbor_direction == Direction::NorthEast {
+                nodes.insert(
+                    NodeRef::North,
+                    *neighbor_hexagon.nodes.get(&NodeRef::SouthWest).unwrap(),
+                );
+                nodes.insert(
+                    NodeRef::NorthEast,
+                    *neighbor_hexagon.nodes.get(&NodeRef::South).unwrap(),
+                );
+                edges.insert(
+                    EdgeRef::NorthEast,
+                    *neighbor_hexagon.edges.get(&EdgeRef::SouthWest).unwrap(),
+                );
+            } else {
+                panic!("Something went wrong");
+            }
```

Comment:
Might using `match` instead of the if/else if conditionals allow you to remove this panic statement?